### PR TITLE
Update semconv

### PR DESCRIPTION
**Pull request by laurit**
Remove `HttpAttributes` and use the constants from `SemanticAttributes` that are now available. Also update messaging conventions.

### CODE REVIEW

File: instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/semconv/messaging/MessagingAttributesGetter.java [hunk 1]
```diff
@@ -25,20 +25,43 @@ public interface MessagingAttributesGetter<REQUEST, RESPONSE> {
...
+  Long getMessageBodySize(REQUEST request);
+
+  @Nullable
+  Long getMessageEnvelopeSize(REQUEST request);
 
   @Nullable
   String getMessageId(REQUEST request, @Nullable RESPONSE response);
 
+  @Nullable
+  String getClientId(REQUEST request);
```
Comment (laurit) [inline instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/semconv/messaging/MessagingAttributesGetter.java:60]: this is currently used only by kafka and was previously set in the kafka extractors

**Comment by SylvainJuge** [comment]
Hi @laurit,

I think it could be worth to also use the semconv bindings to remove the hard-coded ones in [`DistroVersionResourceProvider`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/bc034f3b1b8d66e6c9f5a133407922278bc87086/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/DistroVersionResourceProvider.java#L18-L21) as those have been shipped with semconv 1.22.0.

File: instrumentation/pulsar/pulsar-2.8/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/PulsarClientTest.java [hunk 2]
```diff
@@ -649,14 +660,19 @@ private static List<AttributeAssertion> receiveAttributes(
...
     if (testHeaders) {
       assertions.add(
           equalTo(
               AttributeKey.stringArrayKey("messaging.header.test_message_header"),
               Collections.singletonList("test")));
     }
+    if (isBatch) {
+      assertions.add(
+          satisfies(
+              SemanticAttributes.MESSAGING_BATCH_MESSAGE_COUNT, AbstractLongAssert::isNotNegative));
```
Comment (trask) [inline instrumentation/pulsar/pulsar-2.8/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/PulsarClientTest.java:674]: (same)

File: instrumentation/vertx/vertx-kafka-client-3.6/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/kafka/v3_6/AbstractVertxKafkaTest.java [hunk 3]
```diff
@@ -235,7 +235,10 @@ private static List<AttributeAssertion> batchConsumerAttributes(String topic, St
                 equalTo(SemanticAttributes.MESSAGING_OPERATION, operation),
                 satisfies(
                     SemanticAttributes.MESSAGING_CLIENT_ID,
-                    stringAssert -> stringAssert.startsWith("consumer"))));
+                    stringAssert -> stringAssert.startsWith("consumer")),
+                satisfies(
+                    SemanticAttributes.MESSAGING_BATCH_MESSAGE_COUNT,
+                    AbstractLongAssert::isNotNegative)));
```
Comment (trask) [inline instrumentation/vertx/vertx-kafka-client-3.6/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/kafka/v3_6/AbstractVertxKafkaTest.java:241]: (same)

**Review by trask — APPROVED** [review]

**Pull request merged**